Quantum-Control-Applications/Superconducting/Single Fixed Transmon/Use Case - icognito - Optimized readout with optimal weights/README.md

```diff
@@ -0,0 +1,99 @@
+# Optimized readout with optimal weights
+
+_Author: Kevin Villegas Rosales_
+
+_Demonstrated in **incognito**_
+
+_Important note: The code in this folder is the exact code that we used
+to run the experiment, so it is tailored for the specific software environment
+and set up. Thus, this is code is for inspirational purposes._
+
+## The goal
+
+The goal of this experiment is to optimize the information obtained from the
+readout of a superconducting resonator by deriving the optimal integration weights.
+With the usage of optimal integration weights we maximize the separation of the
+IQ blobs when the ground and excited state are measured.
+
+## The device
+
+The chip contains 5 2D-transoms, where some of them have nearest-neighboor connectivity, 
+with flux tunable lines. For this experiment, we focus only in one transmon with a drive
+to its flux line to put the transom  in the **sweep-spot**. All other transmons are left
+unbiased and signal is send to them.
+
+The readout pulse is `380 ns` long.
+
+## Methods and results
+
+The set of scripts to obtain the readout optimal weights are `TwoStateDiscriminator.py`,
+`TimeDiffCalibrator.py`, and `StateDiscriminator.py`. These three files work together during
+the experiment. Inside these files there are Python classes that are used during
+the training, benchmarking, and later usage in other QUA programs.
+
+First, we will briefly describe each of these files. `TimeDiffCalibrator.py` calculates
+the `time_diff` variable that we used to demodulate analog signals (in post-processing) 
+to match the values of demodulation in the FPGA. `TwoStateDiscriminator.py` contains
+functions to update the `configuration` and to perform measurements with the optimal weights.
+Last, `StateDiscriminator.py` contains the `train` function that from measured analog signals
+calculates the optimal weights.
```

Comment:
As I wrote, we want to use the `qualang_tools`... Otherwise people would download the files from here

Reply:
yes